PowerPointLabs/PowerPointLabs/Models/PowerPointMagnifyingSlide.cs

```diff
@@ -146,9 +145,33 @@ private void AddZoomSlideCroppedPicture(float magnifyRatio = 1.0f)
             cropShape.Select();
             PowerPoint.Selection sel = Globals.ThisAddIn.Application.ActiveWindow.Selection;
             PowerPoint.Shape croppedShape = CropToShape.Crop(zoomSlideCopy, sel, magnifyRatio: magnifyRatio);
-            croppedShape.Cut();
+            try
+            {
+                string tempFilePath = FileDir.GetTemporaryPngFilePath();
+                Utils.GraphicsUtil.ExportShape(croppedShape, tempFilePath);
+                zoomSlideCroppedShapes = _slide.Shapes.AddPicture2(tempFilePath,
+                    Office.MsoTriState.msoFalse,
+                    Office.MsoTriState.msoTrue,
+                    0,
+                    0);
+                croppedShape.Delete();
+                try
+                {
+                    FileDir.DeleteFile(tempFilePath);
+                }
+                catch (Exception)
+                {
+                    // If the file cannot be deleted, we continue without deletion.
+                }
+
+            }
+            catch (Exception)
+            {
+                // Revert to normal copy and pasting if unable to create file.
+                croppedShape.Cut();
+                zoomSlideCroppedShapes = _slide.Shapes.PasteSpecial(PowerPoint.PpPasteDataType.ppPasteMetafilePicture)[1];

```

Comment:
It used to be `PasteSpecial(PowerPoint.PpPasteDataType.ppPastePNG)` in the old code, is `ppPasteMetafilePicture` better? I think it used to throw exceptions for my machine, and I had to use `ppPastePNG` instead.

Reply:
Is there documentation somewhere that states which methods throw exceptions?

Reply:
i used `ppPasteMetaFilePicture` too, it doesn't help with solving the problems